### PR TITLE
[Disease Browser Patch] : Renders graph headers and descriptions only when corresponding data is present

### PR DESCRIPTION
**Pull request by suhana13**
This PR addresses the issue of rendering headers and descriptions only for those graphs where data is available. Here are two pictures of the disease browser which show this in action.
Screenshot 2023-01-31 at 7 45 59 PM
Screenshot 2023-01-31 at 7 46 08 PM

### CODE REVIEW

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 1]
```diff
@@ -326,3 +326,30 @@ export function getDiseaseCommonName(data: GraphNodes): string {
...
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "medicalCondition") {
+      continue;
+    }
+    // check for null or non-existent property values
+    if (_.isEmpty(neighbour.nodes)) {
+      continue;
+    }
+    medicalCondition = neighbour.nodes[0].value;
```
Comment (chejennifer) [inline static/js/biomedical/disease/data_processing_utils.ts:349]: can remove line 337 & update this to `const medicalCondition = neighbour.nodes[0].value`

**Review by chejennifer — COMMENTED** [review]
great, just a few things!

File: static/js/biomedical/disease/page.tsx [hunk 2]
```diff
@@ -34,51 +35,60 @@ import {
...
 
   componentDidUpdate(): void {
-    const diseaseGeneAssociation = getDiseaseGeneAssociation(this.state.data);
-    const diseaseSymptomAssociation = getDiseaseSymptomAssociation(
-      this.state.data
-    );
     drawDiseaseGeneAssocChart(
       "disease-gene-association-chart",
-      diseaseGeneAssociation
+      this.state.diseaseGeneAssociation
```
[the remark this answers is not in the document]
Comment (suhana13) [reply static/js/biomedical/disease/page.tsx:95]: Within the drawDiseaseGeneAssocChart function, we have an if-statement checking for empty data, right at the start. Should I add another check in page.tsx too?

File: static/js/biomedical/disease/page.tsx [hunk 3]
```diff
@@ -34,51 +35,60 @@ import {
...
   }
   render(): JSX.Element {
     const diseaseName = getDiseaseCommonName(this.state.data);
     const diseaseLink = getEntityLink(this.props.dcid);
-    const chemicalCompoundDiseaseTreatment = getCompoundDiseaseTreatment(
-      this.state.data
-    );
-    const chemicalCompoundDiseaseContraindication =
-      getCompoundDiseaseContraindication(this.state.data);
     const diseaseTreatmentColumns = [
```
[the remark this answers is not in the document]
Comment (suhana13) [reply static/js/biomedical/disease/page.tsx]: Done! Changed them to global constants named `DISEASE_TREATMENT_COLUMNS` and `DISEASE_CONTRAINDICATION_COLUMNS`

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 4]
```diff
@@ -326,3 +326,30 @@ export function getDiseaseCommonName(data: GraphNodes): string {
     return formattedDiseaseName;
   }
 }
+/**
+ * Checks whether the stat var for medical condition of disease exists
+ * @param data
+ * @returns boolean indicating if desired stat var exists
+ */
+export function getDiseasePrevalenceID(data: GraphNodes): boolean {
```
[the remark this answers is not in the document]
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts]: Changed the function name to `doesDiseasePrevalenceIDexist` which checks if the ID exists or not!

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 5]
```diff
@@ -326,3 +326,30 @@ export function getDiseaseCommonName(data: GraphNodes): string {
...
+    if (neighbour.property !== "medicalCondition") {
+      continue;
+    }
+    // check for null or non-existent property values
+    if (_.isEmpty(neighbour.nodes)) {
+      continue;
+    }
+    medicalCondition = neighbour.nodes[0].value;
+    if (!_.isEmpty(medicalCondition)) {
+      diseasePrevalenceExists = true;
```
[the remark this answers is not in the document]
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts]: Done!

File: static/js/biomedical/disease/data_processing_utils.ts (hunk 1 above)
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts]: Done!

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 6]
```diff
@@ -326,3 +326,28 @@ export function getDiseaseCommonName(data: GraphNodes): string {
...
   }
 }
+/**
+ * Checks whether the stat var for medical condition of disease exists
+ * @param data
+ * @returns boolean indicating if desired stat var exists
+ */
+export function doesDiseasePrevalenceIDexist(data: GraphNodes): boolean {
+  // sets the default value of the boolean as false
+  if (!data) {
```
Comment (chejennifer) [inline static/js/biomedical/disease/data_processing_utils.ts:336]: probably want to check that data.nodes isn't empty?

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 7]
```diff
@@ -326,3 +326,28 @@ export function getDiseaseCommonName(data: GraphNodes): string {
...
+    }
+    // check for null or non-existent property values
+    if (_.isEmpty(neighbour.nodes)) {
+      continue;
+    }
+    const medicalCondition = neighbour.nodes[0].value;
+    if (!_.isEmpty(medicalCondition)) {
+      return true;
+    }
+    return false;
```
Comment (chejennifer) [inline static/js/biomedical/disease/data_processing_utils.ts:351]: `return false` should probably be outside the for loop?

File: static/js/biomedical/disease/page.tsx [hunk 8]
```diff
@@ -19,78 +19,89 @@
  */
 
 import axios from "axios";
+import _ from "lodash";
 import React from "react";
 
+const DISEASE_TREATMENT_COLUMNS = [
```
Comment (chejennifer) [inline static/js/biomedical/disease/page.tsx:25]: constants should go after the imports

File: static/js/biomedical/disease/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/biomedical/disease/page.tsx:95]: ah I see, ok, that's fine then if there are checks in the draw functions!

**Review by chejennifer — COMMENTED** [review]
thanks for the changes

File: static/js/biomedical/disease/data_processing_utils.ts (hunk 6 above)
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts:336]: Right! Added a check for empty `data.nodes` and empty `data.nodes[0].neighbors`!

File: static/js/biomedical/disease/data_processing_utils.ts (hunk 7 above)
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts:351]: Yes, good catch! Moved it outside the for loop!

File: static/js/biomedical/disease/page.tsx (hunk 8 above)
Comment (suhana13) [reply static/js/biomedical/disease/page.tsx:25]: Done!

File: static/js/biomedical/disease/data_processing_utils.ts [hunk 9]
```diff
@@ -326,3 +326,32 @@ export function getDiseaseCommonName(data: GraphNodes): string {
...
+ * @param data
+ * @returns boolean indicating if desired stat var exists
+ */
+export function doesDiseasePrevalenceIDexist(data: GraphNodes): boolean {
+  // sets the default value of the boolean as false
+  if (!data) {
+    return false;
+  }
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
```
Comment (chejennifer) [inline static/js/biomedical/disease/data_processing_utils.ts:340]: nit: can combine this check with the check on line 336?

File: static/js/biomedical/disease/page.tsx [hunk 10]
```diff
@@ -101,85 +113,109 @@ export class Page extends React.Component<PagePropType, PageStateType> {
       log: false,
       name: "diseaseData",
     };
-    // Render the graphs only when the data is not null
-    if (this.state.data === null) {
```
Comment (chejennifer) [inline static/js/biomedical/disease/page.tsx:105]: nit: I think we should check that the data has been fetched before rendering the page. You can do this by adding one more field in the state 'dataFetched' & checking that field before rendering the page.

**Review by chejennifer — COMMENTED** [review]
thanks for the updates, just a couple more things

File: static/js/biomedical/disease/data_processing_utils.ts (hunk 9 above)
Comment (suhana13) [reply static/js/biomedical/disease/data_processing_utils.ts]: Done!

File: static/js/biomedical/disease/page.tsx (hunk 10 above)
Comment (suhana13) [reply static/js/biomedical/disease/page.tsx:105]: Added the dataFetched field and set it to true when the rest of the fields are fetched!

File: static/js/biomedical/disease/page.tsx [hunk 11]
```diff
@@ -19,78 +19,92 @@
...
     super(props);
-    this.state = { data: null };
+    this.state = {
+      diseaseGeneAssociation: null,
+      diseaseSymptomAssociation: null,
+      chemicalCompoundDiseaseTreatment: null,
+      chemicalCompoundDiseaseContraindication: null,
+      diseaseCommonName: null,
+      diseasePrevalenceIDexists: null,
+      dataFetched: null,
```
Comment (chejennifer) [inline static/js/biomedical/disease/page.tsx:87]: nit: can just set this to false initially?

**Review by chejennifer — APPROVED** [review]
thanks for the changes! one small nit

File: static/js/biomedical/disease/page.tsx (hunk 11 above)
Comment (suhana13) [reply static/js/biomedical/disease/page.tsx]: Right! Set it to false!

**Pull request merged**